### PR TITLE
PR 13757 follow-up: add missing with-discountOverlaps Similarity constructor variants, CHANGES.txt entries (#13845)

### PR DESCRIPTION
**Pull request by cpoerschke**
(cherry picked from commit dab731175c86cc25741f37845136d38a69a9d165)
(cherry picked from commit cbd8b5218addb7712452893cdfe2a3795dc9e05a)

Resolved Conflicts:
	lucene/CHANGES.txt

### CODE REVIEW

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -127,6 +127,8 @@ API Changes
   unifies Byte/FloatVectorValues incorporating RandomAccess* API and introduces
   DocIndexIterator for iterative access in place of direct inheritance from DISI. (Michael Sokolov)
 
+* GITHUB#13845: Add missing with-discountOverlaps Similarity constructor variants. (Pierre Salagnac, Christine Poerschke, Robert Muir)
+
```
Comment (cpoerschke) [inline lucene/CHANGES.txt:131]: Just noting that this is for the 10.0 section and that after this PR is merged (if it is merged) then I'd 10.1 to 10.0 section move the entry correspondingly on the `main` and `branch_10x` branches.

**Comment by cpoerschke** [comment]
Tentatively tagging to 10.0.0 milestone based on https://lists.apache.org/thread/sd0n1kmcz66bt8nspvflh6fbwpb527lq and https://lists.apache.org/thread/pfllhdhdphvgkv1gpngp1r56ydpbt7ov info.

**Review by javanna — APPROVED** [review]

**Comment by javanna** [comment]
I merged this to branch_10_0. Does this also need to backported to 9_12 for the next 9.12.1 patch release?

**Comment by javanna** [comment]
Heads up: I moved the CHANGES entry in main and branch_10x to the 10.0.0 section. See https://github.com/apache/lucene/commit/111fc6f07857f909ad2a253cdcd6e948222ecf63 .

**Comment by cpoerschke** [comment]
> Heads up: I moved the CHANGES entry in main and branch_10x to the 10.0.0 section. ...

Thanks!

> ... Does this also need to backported to 9_12 for the next 9.12.1 patch release?

That would be good, yes, #13891 opened for it.

**Pull request merged**